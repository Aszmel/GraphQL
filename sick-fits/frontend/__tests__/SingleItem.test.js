import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import wait from "waait";
import SingleItem, { SINGLE_ITEM_QUERY } from "../components/SingleItem";
import { MockedProvider } from "react-apollo/test-utils";
import { fakeItem } from "../lib/testUtils";

describe("<SingleItem />", () => {
  it("renders with proper data", async () => {
    const mocks = [
      {
        //when sb makes a request with this query and variables combo
        request: { query: SINGLE_ITEM_QUERY, variables: { id: "123" } },
        //return this fake (mocked) data
        result: {
          data: {
            item: fakeItem()
          }
        }
      }
    ];
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <SingleItem id="123" />
      </MockedProvider>
    );
    // console.log(wrapper.debug());
    //! first c-log will be Loading... <- its waiting for response :)
    expect(wrapper.text()).toContain("Loading...");
    //now we simulating response
    await wait();
    wrapper.update();
    // console.log(wrapper.debug());
    expect(toJSON(wrapper.find("h2"))).toMatchSnapshot();
    expect(toJSON(wrapper.find("img"))).toMatchSnapshot();
    expect(toJSON(wrapper.find("p"))).toMatchSnapshot();
  });

  it("Errors when not found item", async () => {
    const mocks = [
      {
        request: { query: SINGLE_ITEM_QUERY, variables: { id: "123" } },
        result: {
          errors: [{ message: "Item not Found" }]
        }
      }
    ];
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <SingleItem id="123" />
      </MockedProvider>
    );
    await wait();
    wrapper.update();
    const item = wrapper.find('[data-test="graphql-error"]');
    console.log(item.debug());
    expect(item.text()).toContain("Nope!Item not Found");
    expect(toJSON(item)).toMatchSnapshot();
  });
});
