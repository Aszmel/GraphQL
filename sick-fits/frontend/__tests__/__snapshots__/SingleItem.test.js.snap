// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleItem /> Errors when not found item 1`] = `
<p
  data-test="graphql-error"
>
  <strong>
    Nope!
  </strong>
  Item not Found
</p>
`;

exports[`<SingleItem /> renders with proper data 1`] = `
<h2>
  Here is: 
  dogs are best
</h2>
`;

exports[`<SingleItem /> renders with proper data 2`] = `
<img
  alt="dogs are best"
  src="dog.jpg"
/>
`;

exports[`<SingleItem /> renders with proper data 3`] = `
<p>
  dogs
</p>
`;
